From e976adf79e34462acdc2ffe620fadbdd1c0bb592 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Thu, 13 Jul 2017 21:21:35 +0200
Subject: [PATCH 3/5] efi_loader: parameter check
 efi_convert_device_path_to_text

Do not dereference NULL.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_device_path_to_text.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/lib/efi_loader/efi_device_path_to_text.c b/lib/efi_loader/efi_device_path_to_text.c
index c6ac5e52f3..a5ea63300b 100644
--- a/lib/efi_loader/efi_device_path_to_text.c
+++ b/lib/efi_loader/efi_device_path_to_text.c
@@ -36,6 +36,11 @@ uint16_t *efi_convert_device_path_to_text(
 	efi_status_t r;
 	uint16_t *buffer = NULL;
 
+	if (!device_path) {
+		EFI_EXIT(EFI_INVALID_PARAMETER);
+		return NULL;
+	}
+
 	switch (device_path->type) {
 	case DEVICE_PATH_TYPE_MESSAGING_DEVICE:
 		switch (device_path->sub_type) {
-- 
2.11.0

